### ISSUE TITLE
reset leaves behind statefulsets and volume claims etc

### ISSUE THREAD

**Issue by wallyworld** [ubuntu/microk8s#109]
microk8s.reset looks like it deletes deployments, pods, and services.
But if I have created any statefulsets, persistent volume claims, persistent volumes, storage classes, these are not deleted. This includes the storage class created by "enable storage".
This results in stuff being left behind, especially since the statefulset will just restart any pods deleted by the reset command.
Also, I suspect ingress resources are also not deleted.
And daemonsets too?
And custom resource definitions?
Ideally reset would fully clean out the cluster to a totally empty state, including deleting persistent volumes. If we are worried about deleting people's data, then maybe a flag --include-volumes or something could be used to control whether volumes are deleted.

**Comment by wallyworld** [comment]
Juju deploying to microk8s makes use of the resources not currently deleted by reset so adding the missing ones to the wrapper script would be awesome.
- stateful set
- persistent volume claim
- persistent volume
- storage class
- ingress resource
- custom resource definition

**Comment by ktsakalozos** [comment]
Yes we should do a better job in cleaning up resources. A big improvement would be to completely drop the namespaces we can delete. This would release any associated resources. For namespaces we cannot delete we should try to be as thorough as we can.

**Comment by wallyworld** [comment]
@ktsakalozos juju creates a namespace for each model, named the same as the model. When juju deletes a model, it actually just deletes the namespace itself. You could just loop through all namespaces that are not kube or default and delete those. Persistent volumes and Storage classes live outside namespaces so those would need to be deleted separately.
